tooling/jbang/SampleIssueTest.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//DEPS org.hibernate:hibernate-core:5.4.31.Final
+//DEPS junit:junit:4.12
+//DEPS javax.persistence:javax.persistence-api:2.2
+//DEPS org.hibernate.reactive:hibernate-reactive-core:1.0.0.CR4
+//DEPS org.assertj:assertj-core:3.13.2
+//DEPS io.vertx:vertx-pg-client:4.0.3
+//DEPS io.vertx:vertx-db2-client:4.0.3
+//DEPS io.vertx:vertx-mysql-client:4.0.3
+//DEPS io.vertx:vertx-sql-client:4.0.3
+//DEPS io.vertx:vertx-unit:4.0.3
+//
+//
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.hibernate.reactive.provider.ReactiveServiceRegistryBuilder;
+import org.hibernate.reactive.provider.Settings;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.RunWith;
+import org.junit.runner.notification.Failure;
+
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This test class provides a working example of setting up a simple hibernate-reactive test compatible with
+ * hibernate-reactive junit tests.
+ * See <a href="github.com/hibernate/hibernate-reactive/tree/main/hibernate-reactive-core/src/test/java/org/hibernate/reactive">hibernate reactive junit tests</a>
+ *
+ * Developers can copy/paste/edit this test to capture test failures or use-cases and attach an executable class to
+ * issues or other correspondence.
+ *
+ * This test can be executed using the jbang CLI with the command `jbang SampleIssueTest.java`
+ * See <a href="https://github.com/jbangdev/jbang">jbang</a>"
+ * jbang will compile and execute a jar generated based on dependencies defined above in the
+ * file header (i.e.  //DEPS org.hibernate:hibernate-core:5.4.31.Final )
+ *
+ *  >  Customize your JDBC connection properties
+ *  >  Define your specific entity classes and relationships
+ *  >  Replicate your issue and or exception
+ *  >  attach the resulting file or a URL link to it
+ *
+ *  Note this test utilizes a VertxUnitRunner class which provides the hooks to the reactive
+ *  framework for transaction control.
+ *
+ */
+@RunWith(VertxUnitRunner.class)
+public class SampleIssueTest {
+
+	private Mutiny.SessionFactory sessionFactory;
+
+	/**
+	 * Define the configuration parameter values for your use-case
+	 */
+	private Configuration createConfiguration() {
+		Configuration configuration = new Configuration();
+
+		// Use the correct JDBC url
+		configuration.setProperty( Settings.URL, "jdbc:postgresql://localhost:5432/hreact" );
```

Comment:
It might be nice to have the default JDBC strings for the other databases as well (commented for now ).
So that a user doesn't have to look for them if they want to test with a different db.

Reply:
I agree... I even played with adding a DB arg for main that users could `jbang myTest.java mysql`  So we'd have a static map of URL's.  But that might open up having to add a bunch more CLI properties (pwd/user etc...) and seemed out of scope.